### PR TITLE
feat: #990 added 50 dev urls for each cognito client

### PR DESCRIPTION
**Pull request by MCatherine1994**
refs: #990

### CODE REVIEW

File: infrastructure/server/oidc_clients_spar.tf [hunk 1]
```diff
@@ -43,10 +43,10 @@ resource "aws_cognito_user_pool_client" "test_spar_oidc_client" {
   callback_urls                                 = [
     "http://localhost:3000/",
     "https://oidcdebugggersecure-c6af30-dev.apps.gold.devops.gov.bc.ca/",
-    "https://nr-spar-webapp-test-frontend.apps.silver.devops.gov.bc.ca/"
+    "https://nr-spar-test-frontend.apps.silver.devops.gov.bc.ca/"
```
Comment (ianliuwk1019) [inline infrastructure/server/oidc_clients_spar.tf:46]: Hi Catherine, I didn't see this in ticket requirement, wondering, does SPAR request this recent change to remove `webapp` from domain?

**Comment by MCatherine1994** [comment]
> Looks good!
> One question, do you want to generalize the `range(50)` counts into a terraform variable for `50` that can be used everywhere for future clients?

Hi Ian, I was thinking that perviously, but not sure if the client could want different range in the future....

File: infrastructure/server/oidc_clients_spar.tf (hunk 1 above)
Comment (MCatherine1994) [reply infrastructure/server/oidc_clients_spar.tf:46]: yeah, their deployment url for test and prod changed.

**Comment by ianliuwk1019** [comment]
> > Looks good!
> > One question, do you want to generalize the `range(50)` counts into a terraform variable for `50` that can be used everywhere for future clients?
> 
> Hi Ian, I was thinking that perviously, but not sure if the client could want different range in the future....

I think most likely not, but it is a variable that has default value 50. And with that the other part of this issue (github workflow change with Derek), I assume 50 is already sufficiently the upper bound. If client has specific needs, it can certainly use other number.

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**